client/sites-dashboard/components/link-in-bio-banner/link-in-bio-banner.tsx

```diff
@@ -0,0 +1,49 @@
+import { useMobileBreakpoint } from '@automattic/viewport-react';
+import { useSelector } from 'react-redux';
+import { SiteExcerptData } from 'calypso/data/sites/site-excerpt-types';
+import { getPreference } from 'calypso/state/preferences/selectors';
+import { LINK_IN_BIO_BANNER_PREFERENCE } from './link-in-bio-banner-parts';
+import { LinkInBioBanners, BannerType } from './link-in-bio-banners';
+
+type Props = {
+	displayMode: 'row' | 'grid';
+	sites: SiteExcerptData[];
+};
+
+const hasLinkInBioSite = ( sites: SiteExcerptData[] ) => {
+	return Boolean(
+		sites.find( ( site ) => {
+			const option = site.options || {};
+			return option.site_intent === 'link-in-bio';
```

Comment:
We have access to modern JS in Calypso so we would usually do this like 
```suggestion
			return site.option?.site_intent === 'link-in-bio';
```

Just a nitpick, not required 😄

Reply:
For simple null-safe code where the thing that can be null is a leaf object, I still prefer `|| {}`. Where `?.` shines is for complex objects with many potentially null intermediate objects. Using `?.` adds more code to your final bundle. Here's a comparison from the [Babel REPL](https://babeljs.io/repl/#?browsers=defaults%2C%20not%20ie%2011%2C%20not%20ie_mob%2011&build=&builtIns=false&corejs=3.21&spec=false&loose=false&code_lz=MYewdgzgLgBNCGUCmMC8MDeAoGMQAcoBLcCGALm11xxgF8sGtRJYDjS04pEkA6diUgwAPiMx0A3M1KwAbvAA2AVxTpBpPlAAWRMAHNJQA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=env%2Creact%2Cstage-2&prettier=false&targets=&version=7.19.3&externalPlugins=&assumptions=%7B%7D):

![image](https://user-images.githubusercontent.com/6851384/194195625-a42b04ad-9caa-488e-b4f3-513040af18a1.png)

![image](https://user-images.githubusercontent.com/6851384/194195634-38419c40-3f38-4a6c-935b-f2310cf7a523.png)

Reply:
> Using `?.` adds more code to your final bundle

That's a fair point. I'm looking forward to the day when our evergreen bundle won't need to do this!